viewing_party/party.py

```diff
@@ -1,23 +1,147 @@
 # ------------- WAVE 1 --------------------
+from tests.test_constants import MOVIE_TITLE_1, RATING_1
 
+#make a dictioary with three parameters if three of them are truthy
+#if any of them are none function returns none
 def create_movie(title, genre, rating):
-    pass
+    
 
-# -----------------------------------------
+    new_movie=dict([("title",title),("genre", genre),("rating",rating)])
+  
+    
+    if title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+    return new_movie
+
+#This function needs to add movie to the watched list inside the user_data
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+#This function needs to add movie to the watchlist inside the user_data
+def add_to_watchlist(user_data,movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+#If title in watchlist remove and add to watched. Else return user data as is.
+def watch_movie(user_data,title):
+    for i in user_data["watchlist"]:
+        if i["title"]==title:
+            user_data["watched"].append(i)
+            user_data["watchlist"].remove(i)
+    return user_data
+    # 
+# ----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+#calculate the average rating of user watched list. if watched is empty return 0.
+def get_watched_avg_rating(user_data):
+    test_list=user_data["watched"]
+    ress =  [i["rating"] for i in test_list]
+    if len(ress)==0:
+        return 0
+    else:
+        return sum(ress)/len(ress)

```

Comment:
Hmm I'm a little confused by these variable names. Other coders wouldn't be able to tell what information you are collecting in these variables.

Also, the `sum` method has a for loop under the hood. It is doing the same work as your original fo rloop, so let's combine them!

Also, remember to put your guard clause first before making any variables and using up memory space.

```suggestion
    if len(user_data["watched"])==0:
        return 0
    sum = 0
    for movie in user_data["watched"]:
        sum += movie["rating"]
    return sum/len(user_data["watched"])
```